CondaValueError in Microconda setup

Hi, thx for this nice tool, just need some help to let it run...
My use case is to restore packages according to given `environment.yaml` file, so I followed your example, but I am getting its strange error
```
/usr/share/miniconda3/condabin/conda env create -f environment.yml
##[warning]
CondaValueError: prefix already exists: /usr/share/miniconda3

CondaValueError: prefix already exists: /usr/share/miniconda3

##[error]The process '/usr/share/miniconda3/condabin/conda' failed with exit code 1
```
the full run you can find here https://github.com/PyTorchLightning/pytorch-lightning/pull/2412/checks?check_run_id=821315545
also with the cat of actual config file...

--- a/.github/workflows/example-3.yml
+++ b/.github/workflows/example-3.yml
@@ -27,3 +27,23 @@ jobs:
           conda info
           conda list
           printenv | sort
+
+
+  example-3-no-name:
+    name: Ex3-no-name Linux
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        with:
+          activate-environment: anaconda-client-env
+          environment-file: etc/example-environment-no-name.yml
+          python-version: 3.8
+          condarc-file: etc/example-condarc.yml
+          auto-activate-base: false
+          auto-update-conda: true
+      - shell: bash -l {0}
+        run: |
+          conda info
+          conda list
+          printenv | sort

--- a/etc/example-environment-no-name.yml
+++ b/etc/example-environment-no-name.yml
@@ -1,0 +1,3 @@
+dependencies:
+  - black
+  - anaconda-client

--- a/src/setup.ts
+++ b/src/setup.ts
@@ -826,6 +826,7 @@ async function setupMiniconda(
     if (environmentFile) {
       let environmentYaml: TEnvironment;
       let condaAction: string;
+      let activateEnvironmentToUse: string;
       try {
         const sourceEnvironmentPath: string = path.join(
           process.env["GITHUB_WORKSPACE"] || "",
@@ -843,9 +844,10 @@ async function setupMiniconda(
         environmentYaml["name"] !== activateEnvironment
       ) {
         condaAction = "create";
+        activateEnvironmentToUse = environmentYaml["name"];
         utils.consoleLog(`Creating conda environment from yaml file...`);
         core.warning(
-          'The environment name on "environment-file" is not the same as "enviroment-activate"!'
+          'The environment name on "environment-file" is not the same as "enviroment-activate", using "environment-file"!'
         );
       } else if (
         activateEnvironment &&
@@ -853,11 +855,19 @@ async function setupMiniconda(
       ) {
         utils.consoleLog(`Updating conda environment from yaml file...`);
         condaAction = "update";
+        activateEnvironmentToUse = activateEnvironment;
+      } else if (activateEnvironment && environmentYaml["name"] === undefined) {
+        core.warning(
+          'The environment name on "environment-file" is not defined, using "enviroment-activate"!'
+        );
+        condaAction = "update";
+        activateEnvironmentToUse = activateEnvironment;
       } else {
+        activateEnvironmentToUse = activateEnvironment;
         condaAction = "create";
       }
       result = await condaCommand(
-        `env ${condaAction} -f ${environmentFile}`,
+        `env ${condaAction} --file ${environmentFile} --name ${activateEnvironmentToUse}`,
         useBundled,
         useMamba
       );

